### DISCUSSION TITLE
So i tried to run mc but error occurred

### DISCUSSION THREAD

**Discussion by Warlqrd** [FabricMC/fabric-api#1896] [Player support]
---- Minecraft Crash Report ----
// My bad.

Time: 12/11/21, 11:36 AM
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'hecks'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:518)
	at net.minecraft.client.main.Main.main(Main.java:178)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:602)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)
Caused by: net.fabricmc.loader.api.EntrypointException: Exception while loading entries for entrypoint 'main' provided by 'hecks'
	at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:56)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 12 more
Caused by: net.fabricmc.loader.api.LanguageAdapterException: java.lang.ClassNotFoundException: net.fabricmc.example.ExampleMod
	at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:52)
	at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:117)
	at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
	... 13 more
Caused by: java.lang.ClassNotFoundException: net.fabricmc.example.ExampleMod
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:158)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
	... 15 more

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:518)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Advanced Windows 32 Base API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:User Experience Controls Library:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:Crypto API32:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.19041.867 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:DNS Client API DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.19041.1202 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:IP Helper API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Windows NT BASE API Client DLL:10.0.19041.1348 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Windows NT BASE API Client DLL:10.0.19041.1348 (WinBuild.160101.0800):Microsoft Corporation
		MpOav.dll:IOfficeAntiVirus Module:4.18.2110.6 (WinBuild.160101.0800):Microsoft Corporation
		NLAapi.dll:Network Location Awareness 2:10.0.19041.1151 (WinBuild.160101.0800):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.19041.610 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE for Windows:10.0.19041.1320 (WinBuild.160101.0800):Microsoft Corporation
		OleAut32.dll:OLEAUT32.DLL:10.0.19041.985 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Windows Performance Data Helper DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Remote Procedure Call Runtime:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Windows Shell Common Dll:10.0.19041.964 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:Multi-User Windows USER API Client DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft® C Runtime Library:14.28.29913.0 built by: vcwrkspc:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Windows HTTP Services:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:Windows Socket 2.0 32-Bit DLL:10.0.19041.1081 (WinBuild.160101.0800):Microsoft Corporation
		WSOCK32.dll:Windows Socket 32-Bit DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		Wldp.dll:Windows Lockdown Policy:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		amsi.dll:Anti-Malware Scan Interface:10.0.19041.746 (WinBuild.160101.0800):Microsoft Corporation
		bcrypt.dll:Windows Cryptographic Primitives Library:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		bcryptPrimitives.dll:Windows Cryptographic Primitives Library:10.0.19041.1348 (WinBuild.160101.0800):Microsoft Corporation
		clbcatq.dll:COM+ Configuration Catalog:2001.12.10941.16384 (WinBuild.160101.0800):Microsoft Corporation
		combase.dll:Microsoft COM for Windows:10.0.19041.1320 (WinBuild.160101.0800):Microsoft Corporation
		dbgcore.DLL:Windows Core Debugging Helpers:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc.DLL:DHCP Client Service:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc6.DLL:DHCPv6 Client:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		dt_socket.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		fwpuclnt.dll:FWP/IPsec User-Mode API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		gdi32full.dll:GDI Client DLL:10.0.19041.1110 (WinBuild.160101.0800):Microsoft Corporation
		iertutil.dll:Run time utility for Internet Explorer:11.00.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		java.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		java.exe:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		jdwp.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		jemalloc.dll
		jimage.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		jli.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		jna11605381145932332428.dll:JNA native library:6.1.1:Java(TM) Native Access (JNA)
		jvm.dll:Java HotSpot(TM) 64-Bit server VM:17.0.1.0:Oracle Corporation
		kernel.appcore.dll:AppModel API Host:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		lwjgl.dll
		management.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		management_ext.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		msvcp140.dll:Microsoft® C Runtime Library:14.28.29913.0 built by: vcwrkspc:Microsoft Corporation
		msvcp_win.dll:Microsoft® C Runtime Library:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		msvcrt.dll:Windows NT CRT DLL:7.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		mswsock.dll:Microsoft Windows Sockets 2.0 Service Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		napinsp.dll:E-mail Naming Shim Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		ncrypt.dll:Windows NCrypt Router:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		net.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		netutils.dll:Net Win32 API Helpers DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		nio.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		ntdll.dll:NT Layer DLL:10.0.19041.1023 (WinBuild.160101.0800):Microsoft Corporation
		perfos.dll:Windows System Performance Objects DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		pnrpnsp.dll:PNRP Name Space Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		profapi.dll:User Profile Basic API:10.0.19041.844 (WinBuild.160101.0800):Microsoft Corporation
		rasadhlp.dll:Remote Access AutoDial Helper:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		rsaenh.dll:Microsoft Enhanced Cryptographic Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		sechost.dll:Host for SCM/SDDL/LSA Lookup APIs:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		shlwapi.dll:Shell Light-weight Utility Library:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		srvcli.dll:Server Service Client DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		sunmscapi.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		svml.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		ucrtbase.dll:Microsoft® C Runtime Library:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		urlmon.dll:OLE32 Extensions for Win32:11.00.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		vcruntime140_1.dll:Microsoft® C Runtime Library:14.28.29913.0 built by: vcwrkspc:Microsoft Corporation
		verify.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
		win32u.dll:Win32u:10.0.19041.1320 (WinBuild.160101.0800):Microsoft Corporation
		windows.storage.dll:Microsoft WinRT Storage API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		winrnr.dll:LDAP RnR Provider DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		wshbth.dll:Windows Sockets Helper DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		zip.dll:Java(TM) Platform SE binary:17.0.1.0:Oracle Corporation
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:178)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:602)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)

-- System Details --
Details:
	Minecraft Version: 1.18.1
	Minecraft Version ID: 1.18.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.1, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 314678272 bytes (300 MiB) / 1212153856 bytes (1156 MiB) up to 2118123520 bytes (2020 MiB)
	CPUs: 8
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz
	Identifier: Intel64 Family 6 Model 142 Stepping 12
	Microarchitecture: whiskey Lake
	Frequency (GHz): 1.80
	Number of physical packages: 1
	Number of physical CPUs: 4
	Number of logical CPUs: 8
	Graphics card #0 name: Intel(R) UHD Graphics 620
	Graphics card #0 vendor: Intel Corporation (0x8086)
	Graphics card #0 VRAM (MB): 1024.00
	Graphics card #0 deviceId: 0x3ea0
	Graphics card #0 versionInfo: DriverVersion=27.20.100.8853
	Graphics card #1 name: Radeon 520 Series
	Graphics card #1 vendor: Advanced Micro Devices, Inc. (0x1002)
	Graphics card #1 VRAM (MB): 2048.00
	Graphics card #1 deviceId: 0x6665
	Graphics card #1 versionInfo: DriverVersion=27.20.11044.7
	Memory slot #0 capacity (MB): 4096.00
	Memory slot #0 clockSpeed (GHz): 2.67
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 4096.00
	Memory slot #1 clockSpeed (GHz): 2.67
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 14734.24
	Virtual memory used (MB): 9358.21
	Swap memory total (MB): 6656.00
	Swap memory used (MB): 745.02
	JVM Flags: 4 total; -Xdebug -Xnoagent -Xrunjdwp:transport=dt_socket,address=localhost:52091,server=n,suspend=y -XX:+ShowCodeDetailsInExceptionMessages
	Fabric Mods: 
		fabric: Fabric API 0.44.0+1.18
		fabric-api-base: Fabric API Base 0.4.1+b4f4f6cdc8
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.4.0+16d92c47c8
		fabric-biome-api-v1: Fabric Biome API (v1) 6.0.1+ded849a9c8
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.9+3ac43d95c8
		fabric-command-api-v1: Fabric Command API (v1) 1.1.6+3ac43d95c8
		fabric-commands-v0: Fabric Commands (v0) 0.2.5+b4f4f6cdc8
		fabric-containers-v0: Fabric Containers (v0) 0.1.18+d154e2c6c8
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.4.5+6f53a73dc8
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.8+3ac43d95c8
		fabric-data-generation-api-v1: Fabric Data Generation API (v1) 1.0.0+3fec4ad9c8
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.7+43d29571c8
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.4.5+6b21378ac8
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.16+bfa23f17c8
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.6+b4f4f6cdc8
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.10+3ac43d95c8
		fabric-gametest-api-v1: Fabric Game Test API (v1) 1.0.5+03d159cfc8
		fabric-item-api-v1: Fabric Item API (v1) 1.3.0+691a79b5c8
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.3+3ac43d95c8
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.8+c8aba2f3c8
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.6+b4f4f6cdc8
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.10+c15ca335c8
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.8+3ac43d95c8
		fabric-mining-level-api-v1: Fabric Mining Level API (v1) 1.0.3+3ac43d95c8
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.7+b4f4f6cdc8
		fabric-models-v0: Fabric Models (v0) 0.3.3+3ac43d95c8
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.18+3ac43d95c8
		fabric-networking-v0: Fabric Networking (v0) 0.3.5+b4f4f6cdc8
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.11.0+3b82842ec8
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.8+3ac43d95c8
		fabric-particles-v1: Fabric Particles (v1) 0.2.9+526dc1acc8
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.8.5+3ac43d95c8
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.9+3ac43d95c8
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.12+3ac43d95c8
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.7+b4f4f6cdc8
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.4+7242e9d7c8
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.18+3ac43d95c8
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.9+b4f4f6cdc8
		fabric-rendering-v1: Fabric Rendering (v1) 1.10.3+6b21378ac8
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.11+3ac43d95c8
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.7+3ac43d95c8
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.11+3ac43d95c8
		fabric-structure-api-v1: Fabric Structure API (v1) 2.0.8+295197a7c8
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.2.5+3ac43d95c8
		fabric-textures-v0: Fabric Textures (v0) 1.0.9+3ac43d95c8
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.3.4+7de09f55c8
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.5.5+b4f4f6cdc8
		fabricloader: Fabric Loader 0.12.11
		hecks: tp hack ${version}
		java: Java HotSpot(TM) 64-Bit Server VM 17
		minecraft: Minecraft 1.18.1
	Launched Version: Fabric
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: Unknown
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: <unknown>

**Comment by warjort** [comment]
> java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'hecks'!
> Caused by: java.lang.ClassNotFoundException: net.fabricmc.example.ExampleMod

There is something wrong with your "hecks" mod.
It has a fabric.mod.json that refers to a class that does not exist.
